uikit/viewmodels/src/main/java/com/sygic/maps/uikit/viewmodels/navigation/infobar/InfobarViewModel.kt

```diff
@@ -0,0 +1,106 @@
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package com.sygic.maps.uikit.viewmodels.navigation.infobar
+
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.Observer
+import androidx.lifecycle.ViewModel
+import com.sygic.maps.tools.annotations.AutoFactory
+import com.sygic.maps.uikit.viewmodels.common.datetime.DateTimeManager
+import com.sygic.maps.uikit.viewmodels.common.regional.RegionalManager
+import com.sygic.maps.uikit.viewmodels.common.regional.units.DistanceUnit
+import com.sygic.maps.uikit.viewmodels.common.utils.Distance
+import com.sygic.maps.uikit.viewmodels.common.utils.Elevation
+import com.sygic.maps.uikit.viewmodels.common.utils.Time
+import com.sygic.maps.uikit.views.common.extensions.SPACE
+import com.sygic.maps.uikit.views.navigation.infobar.Infobar
+import com.sygic.maps.uikit.views.navigation.infobar.items.InfobarItemsHolder
+import com.sygic.sdk.navigation.NavigationManager
+import com.sygic.sdk.position.PositionManager
+import java.util.*
+import java.util.concurrent.TimeUnit
+
+/**
+ * A [InfobarViewModel] is a basic ViewModel implementation for the [Infobar] view class. It listens
+ * to the [NavigationManager.OnNaviStatsListener] and set appropriate state to the [Infobar] view.
+ * It also listens to the [Infobar] left and right buttons actions.
+ */
+@AutoFactory
+@Suppress("unused", "MemberVisibilityCanBePrivate")
+open class InfobarViewModel internal constructor(
+    private val regionalManager: RegionalManager,
+    private val dateTimeManager: DateTimeManager,
+    private val positionManager: PositionManager,
+    private val navigationManager: NavigationManager
+) : ViewModel(), NavigationManager.OnNaviStatsListener {
+
+    val primaryItemsHolder = MutableLiveData(InfobarItemsHolder(divider = SPACE))
+    val secondaryItemsHolder = MutableLiveData(InfobarItemsHolder.empty)
+
+    private var distanceUnit: DistanceUnit = DistanceUnit.KILOMETERS
+    private val distanceUnitObserver = Observer<DistanceUnit> { distanceUnit = it }
+
+    init {
+        navigationManager.addOnNaviStatsListener(this)
+        regionalManager.distanceUnit.observeForever(distanceUnitObserver)
+    }
+
+    override fun onNaviStatsChanged(
+        distanceToEnd: Int, timeToEndIdeal: Int, timeToEndWithSpeedProfile: Int,
+        timeToEndWithSpeedProfileAndTraffic: Int, routeProgress: Int
+    ) {
+        if (distanceToEnd == 0 && timeToEndIdeal == 0 && timeToEndWithSpeedProfile == 0
+            && timeToEndWithSpeedProfileAndTraffic == 0 && routeProgress == 0
+        ) {
+            return
+        }
+
+        primaryItemsHolder.value = InfobarItemsHolder(
```

Comment:
`onNaviStatsChanged` chodi hodne casto. Neviem ci je az tak fajn vytvarat zakazdym nove `InfobarItemsHolder`-y

Reply:
však to je kvůli té live data notifikaci, jinak by to nefungovalo přeci :(